gtk/src/toga_gtk/screens.py

```diff
@@ -31,10 +31,10 @@ def get_size(self) -> Size:
         return Size(geometry.width, geometry.height)
 
     def get_image_data(self):
-        if "WAYLAND_DISPLAY" in os.environ:
+        if "WAYLAND_DISPLAY" in os.environ:  # pragma: no cover
             # Not implemented on wayland due to wayland security policies.
             self.interface.factory.not_implemented("Screen.get_image_data() on Wayland")
```

Comment:
This is left as blanket "no cover" because the probe prevents fetching an image for the screen if running on Wayland. However, I imagine that's because Toga crashes out if this code is ever actually ran in practice....because the `Screen.as_image()` API in `core` sends the returned `None` in to `toga.Image`. This may not be the best UX...especially since developers may not even know what Wayland is...

Reply:
Agreed it's not ideal UX; returning a dummy image (blank, same size as the screen?) may be preferable. Up to you whether you roll that into this PR, or log it as a standalone issue.

Reply:
I created https://github.com/beeware/toga/issues/2673 to capture this.